src/libstore/build.cc

```diff
@@ -3950,7 +4246,13 @@ void DerivationGoal::registerOutputs()
        outputs, this will fail. */
     {
         ValidPathInfos infos2;
-        for (auto & i : infos) infos2.push_back(i.second);
+        for (auto & [outputName, newInfo] : infos) {
+            /* FIXME: we will want to track this mapping in the DB whether or
```

Comment:
I might have forgotten something obvious, but is there anything preventing us from calling `linkDeriverToPath` without a derivation? (obviously we don't have the `drvPath`, but afaik we can recompute it since derivations are content-addressed)

Reply:
We have a `BasicDerivation` not `Derivation` in the general case. We don't currently have an `unparse` for those, so it's just a matter of bike-shdding whether we want to take the opportunity to switch to JSON or anything else while we're at it.

Reply:
#3958 shows we don't need a new `unparse`, but there still is the question of whether to either:

 - relax foreign key constraint, so we can link without file
 - write drv file to store so we can link with current foreign key constraint

I have no string preference. My only concern is we might want to keep around any mapping in the *build* closure of other mappings, which changes GC semantics.